### PR TITLE
Fix: ensure all connections to the ProgressDialogs's Task the dialog made are disconnected

### PR DESCRIPTION
**Pull request by Ryex**
fix for [this](https://discord.com/channels/1031648380885147709/1031682669945294869/1295280266113454111)

### CODE REVIEW

File: launcher/ui/dialogs/ProgressDialog.cpp [hunk 1]
```diff
@@ -34,6 +34,7 @@
  */
 
 #include "ProgressDialog.h"
+#include <qforeach.h>
```
Comment (Trial97) [inline launcher/ui/dialogs/ProgressDialog.cpp]: Not blocking but what is this header?

**Review by Trial97 — APPROVED** [review]

File: launcher/ui/dialogs/ProgressDialog.cpp (hunk 1 above)
Comment (Ryex) [reply launcher/ui/dialogs/ProgressDialog.cpp]: grr, clangd autoimports

**Review by TheKodeToad — CHANGES_REQUESTED** [review]
https://doc.qt.io/qt-6/qobject.html#dtor.QObject Qt already handles this

**Comment by Ryex** [comment]
> https://doc.qt.io/qt-6/qobject.html#dtor.QObject Qt already handles this

because the ui is explicitly deleted we have to be absolutely sure it's disconnected first.

**Review by TheKodeToad — APPROVED** [review]
Thanks Qt

**Pull request merged**